subt/docker/robotika/ros/robotika/src/drone_height.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+import rospy
+from nav_msgs.msg import *
+from std_msgs.msg import *
+from sensor_msgs.msg import *
+from geometry_msgs.msg import *
+
+import pdb
+    
+HEIGHT = 1.5
+MAX_ANGULAR = 0.7
+lastScanDown = 0.0
+lastScanUp = 0.0
+lastOdom = None
+validOdom = True
+
+def scan_down_callback(scan):
+    global lastScanDown, publisherTwist
+    lastScanDown = scan
+
+def scan_up_callback(scan):
+    global lastScanUp, publisherTwist
+    lastScanUp = scan
+
+def twist_callback(cmd_vel):
+    global HEIGHT, lastScanDown, lastScanUp
+    
+    
+    if lastScanDown.ranges[0] > HEIGHT:
+        cmd_vel.linear.z = -0.1
+    else:
+        cmd_vel.linear.z = 0.1
+    """
+    if lastScanDown.ranges[0] < lastScanUp.ranges[0]:
+        cmd_vel.linear.z = 0.1
+    else:
+        cmd_vel.linear.z = -0.1
+    """
+    #this is because drone can't handle too aggressive rotation
+    if cmd_vel.angular.z > MAX_ANGULAR: 
+        cmd_vel.angular.z = MAX_ANGULAR
+    elif cmd_vel.angular.z < -MAX_ANGULAR:
+        cmd_vel.angular.z = -MAX_ANGULAR
+
+    #if not validOdom:
+    #    #visual odometry error -> turn around and try to relocalize
+    #    cmd_vel.linear.x = 0
+    #    cmd_vel.linear.y = 0
+    #    cmd_vel.angular.z = MAX_ANGULAR
+
+    publisherTwist.publish(cmd_vel)
+
+def odom_callback(odom):
+    global lastOdom, publisherOdom, validOdom
+    if lastOdom <> None and odom.pose.pose.position.x == 0 and odom.pose.pose.position.y == 0:
```

Comment:
For my education: What is the difference between `lastOdom <> None` and `lastOdom is not None`?

Reply:
`<>` can be [overridden](https://docs.python.org/3/reference/datamodel.html#object.__ne__) while `is` cannot.

Reply:
That sounds like `is` is a better choice here.

Reply:
changed to is